query/src/sql/planner/semantic/type_check.rs

```diff
@@ -650,15 +651,20 @@ impl<'a> TypeChecker<'a> {
                 let op = ComparisonOp::try_from(op)?;
                 let (left, _) = self.resolve(left, None).await?;
                 let (right, _) = self.resolve(right, None).await?;
-
+                let mut data_type = BooleanType::new_impl();
+                if left.data_type() == DataTypeImpl::Null(NullType {})
+                    || right.data_type() == DataTypeImpl::Null(NullType {})
+                {
+                    data_type = NullType::new_impl();
+                }
```

Comment:
You can write as:
```rust
let func = FunctionFactory::instance().get(op, args)?;
let result_type = func.return_type();
```
And use the `result_type` as the `data_type`.

Reply:
done